File: data/src/main/java/org/oppia/data/backends/gae/api/TopicService.java
package org.oppia.data.backends.gae.api;

import org.oppia.data.backends.gae.model.Subtopic;
import org.oppia.data.backends.gae.model.Topic;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface TopicService {

  @GET("topic_data_handler/{topic_name}")
  Call<Topic> getTopicByTopicName(@Path("topic_name") String topicName);

}

File: data/src/main/java/org/oppia/data/backends/gae/api/TopicService.kt
package org.oppia.data.backends.gae.api

import org.oppia.data.backends.gae.model.Topic
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface TopicService {

  @GET("topic_data_handler/{topic_name}")
  fun getTopicByTopicName(@Path("topic_name") topicName: String): Call<Topic>

}
